### ISSUE TITLE
Add support for actions from assistive technology

### ISSUE THREAD

**Issue by alice** [servo/servo#46240]
Assistive technology often provides affordances for users to take action on UI elements, as well as providing an alternative presentation of the UI.

AccessKit provides [`ActionRequest`](https://docs.rs/accesskit/0.24.1/accesskit/struct.ActionRequest.html)s, which are created by the platform adapter when the user uses the AT to perform an action, such as "click".

Once we have an `ActionRequest` in the correct pipeline, we can handle it by:

- Ensuring that the accessibility tree is up to date by performing a reflow and accessibility update;
- Retrieving the `AccessibilityNode` corresponding to the `ActionRequest`'s `target_node`;
   - (If `target_node` refers to a node which is no longer in the accessibility tree, we can just drop the action. We need to update the tree _first_ to ensure we don't get an `AccessibilityNode` which no longer has a corresponding DOM node.)
- Using the accessibility node's `OpaqueNode` to retrieve the corresponding DOM node;
- Performing the action on the DOM node, similarly to how we would for WebDriver.

In servoshell, we have an entry point already marked with a `TODO` in [`handle_winit_app_event()`](https://github.com/servo/servo/blob/main/ports/servoshell/desktop/headed_window.rs#L789).

We will need to consider how to design the entry point in the servo embedder API, particularly considering that we will need to ensure the request gets to the pipeline corresponding to its `target_tree`, which may not always be a top-level pipeline.

**Comment by alice** [comment]
ATs often expect browsers to expose a specific structure to support browser-specific functionality. This hasn't previously been documented explicitly, but has been more of a tacit agreement between browser vendors and AT vendors. @spectranaut has filed an issue to track whether these de facto standards should be documented as well: https://github.com/w3c/aria/issues/2890

In the meantime, we'll need to have conversations with other browser and AT vendors to figure out what we need to do on our end to ensure a usable experience for our users.
